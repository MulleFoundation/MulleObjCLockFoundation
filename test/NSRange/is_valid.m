#import <MulleObjC/MulleObjC.h>

#include <stdio.h>
#include <stdint.h>


static void   test_is_valid( NSRange range)
{
   printf( "[%ld - %ld] is %s\n",
               (long) range.location,
               (long) (range.location + range.length),
               MulleObjCRangeIsValid( range) ? "OK" : "INVALID");
}


int   main( void)
{
   NSRange      range;
   NSUInteger   length;

   for( range.location = -2; range.location != 3; range.location++)
      for( range.length = -2; range.length != 3; range.length++)
         test_is_valid( range);

   return( 0);
}


