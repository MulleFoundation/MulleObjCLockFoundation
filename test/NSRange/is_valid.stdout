[-2...-4] is INVALID
[-2...-3] is INVALID
[-2...-2] is INVALID
[-2...-1] is INVALID
[-2...0] is INVALID
[-1...-3] is INVALID
[-1...-2] is INVALID
[-1...-1] is INVALID
[-1...0] is INVALID
[-1...1] is INVALID
[0...-2] is INVALID
[0...-1] is INVALID
[0...0] is OK
[0...1] is OK
[0...2] is OK
[1...-1] is INVALID
[1...0] is INVALID
[1...1] is OK
[1...2] is OK
[1...3] is OK
[2...0] is INVALID
[2...1] is INVALID
[2...2] is OK
[2...3] is OK
[2...4] is OK
